### PR TITLE
1531 download tracking

### PR DESCRIPTION
**Pull request by Castronova**
Added use tracking for: 

* resource create
* resource download
* resource delete

Added tracking variables:  #1577
* ip address
* activity name
* http code
* request method
* user type
* resource info
* email domain

### CODE REVIEW

File: hs_core/models.py [hunk 1]
```diff
@@ -1588,6 +1589,14 @@ def get_irods_storage(self):
         else:
             return IrodsStorage()
 
+    def get_file_size(self):
```
Comment (alexlemann) [inline hs_core/models.py]: Compare this to the size property below.

File: hs_tracking/management/commands/stats.py [hunk 2]
```diff
@@ -137,62 +136,61 @@ def resources_details(self):
...
-        w.writerow(fields)
-
-        today_start = timezone.datetime.now().replace(
-            hour=0,
-            minute=0,
-            second=0,
-            microsecond=0
-        )
+        # w = csv.writer(sys.stdout)
+        # fields = [
```
Comment (alexlemann) [inline hs_tracking/management/commands/stats.py]: Remove comments

File: hs_tracking/management/commands/stats.py [hunk 3]
```diff
@@ -137,62 +136,61 @@ def resources_details(self):
...
-            values = [
-                v.timestamp,
-                uid,
-                v.session.id,
-                v.name,
-                v.type,
-                v.value,
-            ]
-            w.writerow([unicode(v).encode("utf-8") for v in values])
+            # values = [
```
Comment (alexlemann) [inline hs_tracking/management/commands/stats.py]: More comments

File: hs_tracking/signals.py [hunk 4]
```diff
@@ -1,8 +1,11 @@
 from django.contrib.auth.signals import user_logged_in, user_logged_out
 from django.dispatch import receiver
 
-from .models import Session
+from hs_core.signals import *
```
Comment (alexlemann) [inline hs_tracking/signals.py]: Only import what  you need or import hs_core.signals as a thing.

**Comment by Castronova** [comment]
@alexlemann I've addressed your comments and suggestions.  Ready for you to look at again!

**Comment by alexlemann** [comment]
@Castronova Looks like there are some flake8 errors, maybe test failures, and some commented out code in there. Mind looking at the results from Jenkins and spot check the diff?

**Comment by mjstealey** [comment]
@alexlemann - Mind taking another quick peak at this? Should be a rapid review for the upcoming release. Thanks!

File: hs_core/models.py [hunk 5]
```diff
@@ -1588,6 +1589,14 @@ def get_irods_storage(self):
         else:
             return IrodsStorage()
 
+    # def get_file_size(self):
```
Comment (alexlemann) [inline hs_core/models.py]: Remove commented code

**Comment by alexlemann** [comment]
@mjstealey @Castronova one small nit and then :shipit:

**Pull request merged**